dbms/src/Common/LooseBoundedMPMCQueue.h

```diff
@@ -235,12 +306,11 @@ class LooseBoundedMPMCQueue
         queue.pop_back();
         current_auxiliary_memory_usage -= element.memory_usage;
         assert(!queue.empty() || current_auxiliary_memory_usage == 0);
-        writer_head.notifyNext();
         return element.data;
     }
 
     template <typename U>
-    ALWAYS_INLINE void pushFront(U && data)
+    ALWAYS_INLINE bool pushFront(U && data)
```

Comment:
How about adding comments about the return value?

Reply:
Has added comment `If returns true, then notify writers afterward; if false, no need to notify writers.`.